OtripleS.Web.Api/Services/StudentService.Exceptions.cs

```diff
@@ -39,6 +41,18 @@ private async ValueTask<Student> TryCatch(ReturningRetrieveStudentFunction retur
             }
         }
 
+        private IQueryable<Student> TryCatch(ReturningStudentsFunction returningStudentsFunction)
+        {
+            try
+            {
+                return returningStudentsFunction();
+            }
+            catch (SqlException sqlException)
+            {
+                throw CreateAndLogCriticalDependencyException(sqlException);
+            }
```

Comment:
There are few other exceptions that we need to handle here:
1. `DbUpdateException` we log that as a dependency exception
2. `Exception` to handle all other cases we log this one as a service exception